editor/test/integration/scene_test.clj

```diff
@@ -430,28 +432,28 @@
         [:apple-node-id :apple-node-id]
 
         [:tree-node-id]
-        [:tree-node-id :apple-node-id :apple-node-id]
+        [:apple-node-id :apple-node-id :tree-node-id]
 
         [:door-node-id]
-        [:door-node-id :door-handle-node-id]
+        [:door-handle-node-id :door-node-id]
 
         [:house-node-id]
-        [:house-node-id :door-node-id :door-handle-node-id]
+        [:door-node-id :door-handle-node-id :house-node-id]
 
         [:house-node-id :door-handle-node-id]
-        [:house-node-id :door-node-id :door-handle-node-id]
+        [:door-node-id :house-node-id :door-handle-node-id]
 
         [:bucket-node-id]
         [:bucket-node-id]
 
         [:rope-node-id]
-        [:rope-node-id :bucket-node-id]
+        [:bucket-node-id :rope-node-id]
 
         [:well-node-id]
-        [:well-node-id :rope-node-id :bucket-node-id]
+        [:rope-node-id :bucket-node-id :well-node-id]
 
         [:well-node-id :rope-node-id]
-        [:well-node-id :rope-node-id :bucket-node-id]))
+        [:bucket-node-id :well-node-id :rope-node-id]))
```

Comment:
The actually selected nodes are now drawn after their selected children, so they are moved to the end of the lists here.